plugins/http/src/lib.rs

```diff
@@ -14,25 +14,68 @@ pub use error::{Error, Result};
 
 mod commands;
 mod error;
+#[cfg(feature = "cookies")]
+mod reqwest_cookie_store;
 mod scope;
 
 pub(crate) struct Http {
     #[cfg(feature = "cookies")]
-    cookies_jar: std::sync::Arc<reqwest::cookie::Jar>,
+    cookies_jar_path: std::path::PathBuf,
+    #[cfg(feature = "cookies")]
+    cookies_jar: std::sync::Arc<crate::reqwest_cookie_store::CookieStoreMutex>,
 }
 
 pub fn init<R: Runtime>() -> TauriPlugin<R> {
     Builder::<R>::new("http")
         .setup(|app, _| {
+            #[cfg(feature = "cookies")]
+            let (cookies_jar_path, cookies_jar) = {
+                use crate::reqwest_cookie_store::*;
+                use std::fs::File;
+                use std::io::BufReader;
+                use std::sync::Arc;
+
+                let cache_dir = app.path().app_cache_dir()?;
+                std::fs::create_dir_all(&cache_dir)?;
+
+                let path = cache_dir.join("Cookies");
```

Comment:
Just FYI. I have been testing this and this part seems to break when the "Cookies" did not previously exist.

I changed it so it starts with an empty array in case the file is new:
```
let path = cache_dir.join("Cookies");
let file_exists = path.exists();
let mut file = File::options()
    .create(true)
    .append(true)
    .read(true)
    .open(&path)?;

if !file_exists {
    // Initialize the file with an empty array
    use std::io::Write;
    file.write_all(b"[]")?;
}
```

Reply:
nice catch!